### ISSUE TITLE
create_modal or create_view ? path to custom create modal ?

### ISSUE THREAD

**Issue by DonDiegoAA** [webfactor/laravel-backpack-instant-fields#28]
Hi,

thank you for this package.
In My project, I have purchases linked to supplier (1-n).

I have inserted an instant field into Purchase to trigger Suppliers.

But, my suppliers  views (create and edit) are custom views (I needed to add some js here).

The create_modal and edit_modal options are made for me, if I understood correctly, but I can't figure out how to use it.

In my SupplierCrudController, I have these lines
```
$this->crud->setCreateView('vendor.backpack.crud.supplier-create');
$this->crud->setEditView('vendor.backpack.crud.supplier-edit');
```

So I tried 
```
'on_the_fly'           => [
                'entity'        => 'supplier', // e. g. user, contact, company etc...
                'create_modal'  => 'vendor.backpack.crud.modal.supplier-create',
                'edit_modal'    => 'vendor.backpack.crud.modal.supplier-edit'
]
```

Result: 
GET http://myproject.localhost/admin/purchase/vendor.backpack.crud.modal.supplier-create 404 (Not Found)

I tried many others path types, without best results.

In the doc, you write also
```
'on_the_fly' => [
    'entity' => 'entity',
    'create_view => 'route/to/modal/html'
]
```

Is create_view  another option, or is this a typo ?

**Comment by tswonke** [comment]
Thanks for the hint - I just updated the inconsistency of the readme.

I will try to strip down some things:

1. IMO the modal should be automatically generated (even if you have set a custom create or edit blade in your `SupplierCrudController`). Did you try without `create_modal` and `edit_modal` attributes?

2. If you really need to use custom modals:
Please be sure to provide a valid url (relative or absolute)! You just used a view-string and this does not work. On loading the modal a request (to the provided url) is fired and the resulting HTML will be filled in the modal.
So you will have to build up the whole modal by yourself!

**Comment by DonDiegoAA** [comment]
Thank you for your reply.

Without create_modal or edit_modal, the form loads perfectly.

But if I created custom create_view and edit_view for this model, it's because I needed to add some javascript (some fields must be visible / hidden according to a checkbox).
And this javascript is not included in the modal, which is normal I think.

If I add javascript directly to your code (\vendor\webfactor\laravel-backpack-instant-fields\resources\views\modal\create.blade.php or edit.blade.php), it works as expected. But it's bad practice.

So I copy your files into my resources folder:
\vendor\webfactor\laravel-backpack-instant-fields\resources\views\modal\create.blade.php (and edit) =>
\resources\views\vendor\backpack\crud\modal\supplier-create.blade.php (and edit)
and I add the javascript in my files.

I'm now trying to use create_modal and edit_modal to call my modal, but I can not find what I have to put as a path for these variables.

Do I have to create a method that returns the code of my modal, and a route to call that method. And this is the route I should put in create_modal?
I'm going to try. But then it would be nice to be able to indicate in create_modal the path to the view, and not the path to a method that returns the view, if possible.

**Comment by tswonke** [comment]
Can you please post an example of the javascript you want to use in the fields? I would like to test whether there is a way to make it work out of the box.

I got your point on defining the view, I will think about it.
For now you could overwrite the `ajaxCreate()` and `ajaxEdit()` methods in your `SupplierCrudControlller` (s. Line 72/91 of `HandlesAjaxRequest.php` in our package. There you would just have to switch the used view.

**Comment by DonDiegoAA** [comment]
Here is the code of my custom create view. No change in the html part comparing to backpack view, only added @push ('after_scripts') ... @endpush

```
@extends('backpack::layout')

@section('header')
	<section class="content-header">
	  <h1>
        <span class="text-capitalize">{!! $crud->getHeading() ?? $crud->entity_name_plural !!}</span>
        <small>{!! $crud->getSubheading() ?? trans('backpack::crud.add').' '.$crud->entity_name !!}.</small>
	  </h1>
	  <ol class="breadcrumb">
	    <li><a href="{{ url(config('backpack.base.route_prefix'), 'dashboard') }}">{{ trans('backpack::crud.admin') }}</a></li>
	    <li><a href="{{ url($crud->route) }}" class="text-capitalize">{{ $crud->entity_name_plural }}</a></li>
	    <li class="active">{{ trans('backpack::crud.add') }}</li>
	  </ol>
	</section>
@endsection

@section('content')
@if ($crud->hasAccess('list'))
	<a href="{{ url($crud->route) }}" class="hidden-print"><i class="fa fa-angle-double-left"></i> {{ trans('backpack::crud.back_to_all') }} <span>{{ $crud->entity_name_plural }}</span></a>
@endif

<div class="row m-t-20">
	<div class="{{ $crud->getCreateContentClass() }}">
		<!-- Default box -->

		@include('crud::inc.grouped_errors')

		  <form method="post"
		  		action="{{ url($crud->route) }}"
				@if ($crud->hasUploadFields('create'))
				enctype="multipart/form-data"
				@endif
		  		>
		  {!! csrf_field() !!}
		  <div class="col-md-12">

		    <div class="row display-flex-wrap">
		      <!-- load the view from the application if it exists, otherwise load the one in the package -->
		      @if(view()->exists('vendor.backpack.crud.form_content'))
		      	@include('vendor.backpack.crud.form_content', [ 'fields' => $crud->getFields('create'), 'action' => 'create' ])
		      @else
		      	@include('crud::form_content', [ 'fields' => $crud->getFields('create'), 'action' => 'create' ])
		      @endif
		    </div><!-- /.box-body -->
		    <div class="">

                @include('crud::inc.form_save_buttons')

		    </div><!-- /.box-footer-->

		  </div><!-- /.box -->
		  </form>
	</div>
</div>
@endsection

@push('after_scripts')
<script>

jQuery(document).ready(function($){
	$("select[name='city_id']").on('select2:select', function (e) {
		var data = e.params.data;

		$.ajax({
            url: '/ajax/city/'+data.id,
            type: 'GET',
            success: function(result) {
				
				$("select[name='language_id']").val(result.language_id); 
				$("select[name='language_id']").trigger('change');
				
				$("select[name='region_id']").val(result.region_id); 
				$("select[name='region_id']").trigger('change');
				
				$("input[name='postal_code']").val(result.postal_code); 
              },
              error: function(result) {
                  // Show an alert with the result
                  //nothing to do
              }
          });
	});

	toggleForeignerFields = function() {
		if( $("input[name='foreigner']").is(':checked') ) {
			$(".is_not_foreigner").hide();
			$(".is_foreigner").show();

		}else {
			$(".is_foreigner").hide();
			$(".is_not_foreigner").show();
		}

	}

	$("input[name='foreigner']").change(function() {
              toggleForeignerFields();
	}); 

	toggleForeignerFields();
});
</script>
@endpush
```

**Comment by tswonke** [comment]
@DonDiegoAA 

I just released v2.2 where you can define `create_modal_view`, `edit_modal_view` and `delete_modal_view`, so you don't have to overwrite `ajaxCreate()` (and the other methods) anymore or even custom routes.

Hope this will work, I didn't make detailed tests.
